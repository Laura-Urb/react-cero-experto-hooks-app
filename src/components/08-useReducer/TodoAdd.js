import React from 'react';
import { useForm } from '../../hooks/useForm';

export const TodoAdd = ({ handleAddTodo }) => {
  const [{ descripcion }, handleInputChange, reset] = useForm({
    descripcion: '',
  });
  const handleSubmit = (e) => {
    e.preventDefault();
    if (descripcion.trim().length <= 1) return;

    const newTodo = {
      id: new Date().getTime().toString(),
      desc: descripcion,
      done: false,
    };

    handleAddTodo(newTodo);
    reset();
  };
  return (
    <>
      <h4>Agregar TODO</h4>
      <hr />
      <form onSubmit={handleSubmit}>
        <input
          type=''
          name='descripcion'
          value={descripcion}
          placeholder='Aprender...'
          autoComplete='off'
          className='form-control'
          onChange={handleInputChange}
        />
        <button type='submit' className='btn btn-outline-primary mt-1 w-100'>
          Agregar
        </button>
      </form>
    </>
  );
};
