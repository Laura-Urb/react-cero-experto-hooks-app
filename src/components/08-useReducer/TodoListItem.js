import React from 'react';

export const TodoListItem = ({ todo, i, handleDelete, handleToogle }) => {
  return (
    <li className='list-group-item'>
      <p
        className={`${todo.done && 'complete'}`}
        onClick={() => handleToogle(todo.id)}
      >
        {i + 1}. {todo.desc}
      </p>
      <button onClick={() => handleDelete(todo.id)} className='btn btn-danger'>
        Borrar
      </button>
    </li>
  );
};
