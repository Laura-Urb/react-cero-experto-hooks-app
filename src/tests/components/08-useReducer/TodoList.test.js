import { TodoList } from '../../../components/08-useReducer/TodoList';
import React from 'react';
import { shallow } from 'enzyme';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Pruebas TodoList', () => {
  const handleDelete = jest.fn();
  const handleToogle = jest.fn();
  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleDelete={handleDelete}
      handleToogle={handleToogle}
    />
  );
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de tener dos TodoListItem', () => {
    expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
    expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(
      expect.any(Function)
    );
  });
});
