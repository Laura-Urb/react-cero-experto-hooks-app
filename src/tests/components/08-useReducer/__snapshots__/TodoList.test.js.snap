// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas TodoList debe de mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "AAA",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "BBB",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
